### DISCUSSION TITLE
why v-if type assert not work well?

### DISCUSSION THREAD

**Discussion by lovetingyuan** [vuejs/language-tools#996] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/7310471/156156717-c462af37-f5dd-4ebb-b986-a8d3cfff3d7b.png)

**Comment by lovetingyuan** [comment, marked as answer]
v-for can not work with v-if...

**Reply by johnsoncodehk** [reply]
It can, but please note that if you config `"experimentalCompatMode": 2`, `v-for` has higher priority than `v-if`.

**Reply by lovetingyuan** [reply]
Thank you
